lib/org/openpsa/sales/sales/pdf.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * @package org.openpsa.sales
+ * @copyright CONTENT CONTROL http://www.contentcontrol-berlin.de/
+ * @license http://www.gnu.org/licenses/gpl.html GNU General Public License
+ */
+
+/**
+ * PDF Manager
+ *
+ * @package org.openpsa.sales
+ */
+class org_openpsa_sales_sales_pdf
+{
+    /**
+     * @var org_openpsa_sales_salesproject_dba
+     */
+    private $salesproject;
+
+    public function __construct(org_openpsa_sales_salesproject_dba $salesproject)
+    {
+        $this->salesproject = $salesproject;
+    }
+
+    /**
+     * @return midcom_db_attachment|NULL
+     */
+    public function get_attachment($autocreate = false)
+    {
+        $pdf_files = org_openpsa_helpers::get_dm2_attachments($this->salesproject,"pdf_file");
+        if (!empty($pdf_files)) {
+            return reset($pdf_files);
+        }
+        if (!$autocreate) {
+            return null;
+        }
+        return $this->render_and_attach();
+    }
+
+    public function get_button_options()
+    {
+       if ($attachment = $this->get_attachment()) {
+
+            // check if auto generated parameter is same as md5 in current-file
+            // if not the file was manually uploaded
+            if ($checksum = $attachment->get_parameter('org.openpsa.sales', 'auto_generated')) {
+                $blob = new midgard_blob($attachment->__object);
+                if ($checksum !== md5_file($blob->get_path())) {
+                    $message = 'current pdf file was manually uploaded shall it be replaced ?';
```

Comment:
I don't think we even have an option for manual upload currently, so this is probably unused